src/Controller/Admin/ContactExtensionController.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Admin;
+
+use App\Entity\Contact;
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\View\ViewHandlerInterface;
+use HandcraftedInTheAlps\RestRoutingBundle\Controller\Annotations\RouteResource;
+use HandcraftedInTheAlps\RestRoutingBundle\Routing\ClassResourceInterface;
+use Sulu\Bundle\ContactBundle\Controller\ContactController;
+use Sulu\Component\Rest\AbstractRestController;
+use Sulu\Component\Security\SecuredControllerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * @RouteResource("contact-extension")
+ */
+class ContactExtensionController extends AbstractRestController implements ClassResourceInterface, SecuredControllerInterface
+{
+    private EntityManagerInterface $entityManager;
+    private ContactController $contactController;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        ContactController $contactController,
+        ViewHandlerInterface $viewHandler,
+        ?TokenStorageInterface $tokenStorage = null
+    ) {
+        $this->entityManager = $entityManager;
+        $this->contactController = $contactController;
+
+        parent::__construct($viewHandler, $tokenStorage);
+    }
+
+    public function getAction(int $id): Response
+    {
+        $contact = $this->entityManager->getRepository(Contact::class)->find($id);
+        if (!$contact) {
+            throw new NotFoundHttpException();
+        }
+
+        return $this->handleView($this->view($this->getDataForEntity($contact)));
+    }
+
+    public function putAction(Request $request, int $id): Response
+    {
+        $contact = $this->entityManager->getRepository(Contact::class)->find($id);
+        if (!$contact) {
+            throw new NotFoundHttpException();
+        }
+
+        $this->mapDataToEntity($request->request->all(), $contact);
+        $this->entityManager->flush();
+
+        return $this->handleView($this->view($this->getDataForEntity($contact)));
+    }
+
+    public function deleteAction(int $id): Response
+    {
+        return $this->contactController->deleteAction($id);
```

Comment:
I would forward the request so the requestStack is correctly called and a listener which does listen to `DeleteAction` route would be correctly called.

https://symfony.com/doc/current/controller/forwarding.html